components/layout_2020/table/mod.rs

```diff
@@ -2,12 +2,66 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-//! HTML Tables (╯°□°)╯︵ ┻━┻.
+//! # HTML Tables (╯°□°)╯︵ ┻━┻
 //!
-//! See <https://html.spec.whatwg.org/multipage/table-processing-model> and
-//! <https://drafts.csswg.org/css-tables>.  This is heavily based on the latter specification, but
-//! note that it is still an Editor's Draft, so there is no guarantee that what is implemented here
-//! matches other browsers or the current specification.
+//! This implementation is based on the [table section of the HTML 5
+//! Specification](https://html.spec.whatwg.org/multipage/#tables), the draft [CSS Table Module Level
+//! 3](https://drafts.csswg.org/css-tables) and the [LayoutNG implementation of
+//! tables](https://source.chromium.org/chromium/chromium/src/third_party/+/main:blink/renderer/core/layout/table)
+//! in Blink. In general, the draft specification differs greatly from what other browsers do, so we
+//! generally follow LayoutNG when in question.
+//!
+//! Table layout is divided into two phases:
+//!
+//! 1. Box Tree Construction
+//! 2. Fragment Tree Construction
+//!
+//! ## Box Tree Construction
+//!
+//! During box tree construction, table layout (`construct.rs`) will traverse the DOM and construct
+//! the basic box tree representation of a table, using the structs defined in this file ([`Table`],
+//! [`TableTrackGroup`], [`TableTrack`], etc). When processing the DOM, elements are handled
+//! differently depending on their `display` value. For instance, an element with `display:
+//! table-cell` is treated as a table cell. These display values are automatically set for HTML
+//! elements by the user agent stylesheet.
+//!
+//! Inside each table is an array of [`TableSlot`]. A table slot represents either a cell, a cell
+//! location occupied by a cell originating from another place in the table, or is empty. In
+//! addition, when there are table model errors, a slot may spanned by more than one cell. In those
+//! cases, the top-left-most cell wins.
+//!
+//! During processing, the box tree construction agorithm will also fix up table structure, for
+//! instance, creating anonymous rows for lone table cells and putting other anonymous content into
+//! cells. In addition, flow layout will collect table elements outside of tables and create
```

Comment:
The content that is put into cells isn't typically anonymous.

```suggestion
//! instance, creating anonymous rows for lone table cells and putting non-tabular content into
//! anonymous cells. In addition, flow layout will collect table elements outside of tables and create
```